crates/storage/db-models/src/client_version.rs

```diff
@@ -3,23 +3,48 @@
 use reth_codecs::{add_arbitrary_tests, Compact};
 use serde::{Deserialize, Serialize};
 
+/// Wrapper around `String`
+#[derive(Clone, Eq, PartialEq, Debug, Default, Serialize, Deserialize)]
+pub struct CompactString(String);
+
 /// Client version that accessed the database.
 #[derive(Clone, Eq, PartialEq, Debug, Default, Serialize, Deserialize)]
 #[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
 #[add_arbitrary_tests(compact)]
 pub struct ClientVersion {
     /// Client version
-    pub version: String,
+    pub version: CompactString,
     /// The git commit sha
-    pub git_sha: String,
+    pub git_sha: CompactString,
     /// Build timestamp
-    pub build_timestamp: String,
+    pub build_timestamp: CompactString,
 }
 
 impl ClientVersion {
     /// Returns `true` if no version fields are set.
     pub fn is_empty(&self) -> bool {
-        self.version.is_empty() && self.git_sha.is_empty() && self.build_timestamp.is_empty()
+        self.version.0.is_empty() && self.git_sha.0.is_empty() && self.build_timestamp.0.is_empty()
+    }
+}
+
+impl From<String> for CompactString {
+    fn from(s: String) -> Self {
+        CompactString(s)
+    }
+}
+
+impl Compact for CompactString {
```

Comment:
you can move this to the reth-codecs crate, where the trait is defined, then you don't need the wrapper